app/packages/looker/src/worker/deserializer.ts

```diff
@@ -2,8 +2,15 @@ import { deserialize } from "../numpy";
 
 export const DeserializerFactory = {
   Detection: (label, buffers) => {
+    let serializedMask: string;
     if (typeof label?.mask === "string") {
-      const data = deserialize(label.mask);
+      serializedMask = label.mask;
+    } else if (typeof label?.mask?.$binary?.base64 === "string") {
+      serializedMask = label.mask.$binary.base64;
+    }
+
+    if (serializedMask) {
+      const data = deserialize(serializedMask);
```

Comment:
I wouldn't be upset if you removed all the code duplication.

```typescript
    const data = deserializeProp(label, "mask");
    if (data) {
...

function deserializeProp(label, prop) {
  let serializedMask: string | null = null;
  if (typeof label?.[prop] === "string") {
      serializedMask = label[prop];
    } else if (typeof label?.[prop]?.$binary?.base64 === "string") {
      serializedMask = label[prop].$binary.base64;
    }
  }
  if (!serializedMask) {
    return null;
  }
  return deserialize(serializedMask);
}
```

Reply:
yeah agreed - I started to and then it was more code than it was worth so I got lazy. I'll refactor.

Reply:
Thanks!